web/src/engine/websites/Remanga.ts

```diff
@@ -0,0 +1,146 @@
+     */
+    public async UpdateToken() {
+        try {
+            this.#token = await FetchWindowScript(new Request(this.clientURI), `(async () => (await cookieStore.get('token'))?.value ?? undefined)();`, 750);
+        } catch (error) {
+            console.warn('UpdateToken()', error);
+            this.#token = null;
+        }
+    }
+
+    /**
+     * Determine the _Bearer_ extracted from the current token and add it as authorization header to the given {@link init} headers (replacing any existing authorization header).
+     * In case the _Bearer_ could not be extracted from the current token the authorization header will not be added/replaced.
+     */
+    public async ApplyAuthorizationHeader(init: HeadersInit): Promise<HeadersInit> {
+        const headers = new Headers(init);
+        if (this.#token) {
+            headers.set('Authorization', 'Bearer ' + this.#token);
+        }
+        return headers;
+    }
+}
+
+@Common.ImageAjax(true)
+export default class extends DecoratableMangaScraper {
+
+    private readonly apiUrl = 'https://api.remanga.org/api/v2/';
+    #tokenProvider: TokenProvider;
+
+    public constructor() {
+        super('remanga', 'Remanga', 'https://remanga.org', Tags.Media.Manga, Tags.Media.Manhwa, Tags.Media.Manhua, Tags.Language.Russian, Tags.Source.Aggregator, Tags.Accessibility.RegionLocked);
+    }
+
+    public override get Icon() {
+        return icon;
+    }
+
+    public override async Initialize(): Promise<void> {
+        this.#tokenProvider = new TokenProvider(this.URI);
+        this.#tokenProvider.UpdateToken();
+    }
+
+    public override ValidateMangaURL(url: string): boolean {
+        return new RegExpSafe(`^${this.URI.origin}/manga/[^/]+/main$`).test(url);
+    }
+
+    public override async FetchManga(provider: MangaPlugin, url: string): Promise<Manga> {
+        const { dir, main_name, secondary_name } = await this.GetMangaDetails(new URL(url).pathname.split('/').at(-2));
+        return new Manga(this, provider, dir, main_name ?? secondary_name);
+    }
+
+    public override async FetchMangas(provider: MangaPlugin): Promise<Manga[]> {
+        const mangaList: Manga[] = [];
+        for (let page = 1, run = true; run; page++) {
+            await Delay(500);
+            const mangas = await this.GetMangasFromPage(page, provider);
+            mangaList.push(...mangas);
+            run = mangas.length > 0 && page < 1000; //website api is limited to 999 pages
+        }
+        return mangaList;
+    }
+
+    private async GetMangasFromPage(page: number, provider: MangaPlugin): Promise<Manga[]> {
+        const { results } = await this.FetchAPI<APIResult<APIManga[]>>(`./search/catalog/?count=30&ordering=score&page=${page}`);
+        return results.map(manga => new Manga(this, provider, manga.dir, manga.main_name ?? manga.secondary_name));
+    }
+
+    private async GetMangaDetails(slug: string): Promise<APIManga> {
```

Comment:
I don't see much benefit for this intermediate function instead of using `FetchAPI` directly